rs/ethereum/cketh/minter/src/eth_logs/mod.rs

```diff
@@ -378,6 +441,30 @@ impl TryFrom<LogEntry> for ReceivedEvent {
     }
 }
 
+fn parse_data_into_32_bytes_words<const N: usize>(
```

Comment:
Nit: Since it's 32-**byte** words, how about calling the function `parse_data_into_32_byte_words`?

Reply:
I though one could alternatively use `32-byte words` (with hyphen) or `32 bytes words` (without hyphen). Are they not both correct?

Reply:
Not related to the above comments, but this is a very cool use case of dependent types, namely, value depending on type. I didn't know Rust could do this!

Reply:
I didn't manage to track down an authoritative explanation of the exact rule but a hyphen is always used when binding words that modify a noun (such as a "two-year-old child" where "two-year-old" functions as an adjective, see [here](https://ell.stackexchange.com/questions/43196/difference-between-two-years-old-and-two-year-old)). Similarly, nouns that modify nouns are also considered adjectives and therefore _usually_ appear in singular form, see [here](https://english.stackexchange.com/questions/258111/two-plural-nouns-in-a-row). An interesting exception is "arms race", where "arms" is plural to disambiguate.

In short, "32-byte words" is certainly the more commonly accepted form.
I claim that "32 bytes words" would sound unnatural to most native speakers.

Reply:
Thanks a lot Thomas for this explanation! Let's remove the `s` then!